### PR TITLE
[PREVIEW] moving to 3.4 compaitable api

### PR DESCRIPTION
**Pull request by schernolyas**
HI!
 The PR contains changes that required for migration to 3.4 api.
[3.4 doc](http://mongodb.github.io/mongo-java-driver/3.4/)

### CODE REVIEW

**Comment by Sanne** [comment]
Hi @schernolyas,
sorry we can't integrate changes when they include so many unrelated changes.

For example I see many cases in which you re-formatted the code. We can't allow that to happen in a large OSS project, as if everyone would that that, every contributor would have massive amounts of conflicts to solve.

It's also much easier for who has to review your code, as with such a large diff I just can't see what you changed exactly.

Can you please make sure your patches are small and to the point?

Thanks!

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/MongoDBIndexTest.java [hunk 1]
```diff
@@ -59,17 +63,18 @@ public void testSuccessfulIndexCreation() throws Exception {
...
+				+ " , 'key' : { '_fts' : 'text' , '_ftsx' : 1} , 'name' : 'author_name_text_idx' , 'ns' : 'ogm_test_database.T_POEM' , 'weights' : { 'author' : 2, 'name' : 5} , 'default_language' : 'fr' , 'language_override' : 'language' , 'textIndexVersion' : 3}",
 				indexMap.get( "author_name_text_idx" ).toString() );
 
 		session.close();
 	}
 
 	@Override
 	protected Class<?>[] getAnnotatedClasses() {
-		return new Class<?>[] { Poem.class };
+		return new Class<?>[]{ Poem.class };
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/MongoDBIndexTest.java]: An example of changes which should be avoided.

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/Poem.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
 		@Index(columnList = "author, name", name = "author_name_text_idx"),
 		@Index(columnList = "", name = "index_with_no_keys_idx"),
 		@Index(columnList = "name, author"), // index with no name: the name will be generated by ORM
-} )
+})
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/Poem.java]: An example of changes which should be avoided.

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/MongoDBIndexTest.java [hunk 3]
```diff
@@ -59,17 +63,18 @@ public void testSuccessfulIndexCreation() throws Exception {
 	public void testSuccessfulTextIndexCreation() throws Exception {
 		OgmSession session = openSession();
 
-		Map<String, DBObject> indexMap = getIndexes( session.getSessionFactory(), COLLECTION_NAME );
+		Map<String, Document> indexMap = getIndexes( session.getSessionFactory(), COLLECTION_NAME );
 		assertThat( indexMap.size() ).isEqualTo( 6 );
 
-		assertJsonEquals( "{ 'v' : " + VERSION + " , 'key' : { '_fts' : 'text' , '_ftsx' : 1} , 'name' : 'author_name_text_idx' , 'ns' : 'ogm_test_database.T_POEM' , 'weights' : { 'author' : 2, 'name' : 5} , 'default_language' : 'fr' , 'language_override' : 'language' , 'textIndexVersion' : 3}",
+		assertJsonEquals( "{ 'v' : " + VERSION
+				+ " , 'key' : { '_fts' : 'text' , '_ftsx' : 1} , 'name' : 'author_name_text_idx' , 'ns' : 'ogm_test_database.T_POEM' , 'weights' : { 'author' : 2, 'name' : 5} , 'default_language' : 'fr' , 'language_override' : 'language' , 'textIndexVersion' : 3}",
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/MongoDBIndexTest.java]: An example of changes which should be avoided.

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/loading/LoadSelectedColumnsCollectionTest.java [hunk 4]
```diff
@@ -202,39 +196,36 @@ private Tuple getTuple(String collectionName, String id, List<String> selectedCo
 	protected void configure(Map<String, Object> settings) {
 		settings.put(
 				DocumentStoreProperties.ASSOCIATIONS_STORE,
-				AssociationStorageType.ASSOCIATION_DOCUMENT
-		);
+				AssociationStorageType.ASSOCIATION_DOCUMENT );
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/loading/LoadSelectedColumnsCollectionTest.java]: An example of changes which should be avoided.

**Comment by schernolyas** [comment]
Hi @Sanne 

Unfortunately, migration contains a lot of changes. 
About unrelated changes .... Ok. I will try to revert there changes. And  I will fix it in next commits

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/mapping/EmbeddableIdMappingTest.java [hunk 5]
```diff
@@ -64,7 +62,7 @@ public void multiColumnEmbeddableIdMapping() {
 		OgmSession session = openSession();
 		Transaction transaction = session.beginTransaction();
 
-		assertDbObject(
+		assertDocument(
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/mapping/EmbeddableIdMappingTest.java]: An example of changes which should be avoided.

To be fair, the change itself isn't bad, it's just much easier for all if you do a method rename in a dedicated commit.

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/options/associationstorage/AssociationStorageOptionTest.java [hunk 6]
```diff
@@ -33,16 +33,17 @@ public void testAssociationStorageMappingOption() throws Exception {
 		ConfigurationContext configurationContext = new ConfigurationContextImpl( context );
 
 		new MongoDB().getConfigurationBuilder( configurationContext )
-			.entity( ExampleForMongoDBMapping.class )
+				.entity( ExampleForMongoDBMapping.class )
 				.property( "content", ElementType.FIELD )
-					.associationStorage( AssociationStorageType.ASSOCIATION_DOCUMENT );
+				.associationStorage( AssociationStorageType.ASSOCIATION_DOCUMENT );
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/options/associationstorage/AssociationStorageOptionTest.java]: The previous indentation had a specific meaning. Please be careful in not throwing such things away.

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/options/associationstorage/AssociationStorageOptionTest.java [hunk 7]
```diff
@@ -33,16 +33,17 @@ public void testAssociationStorageMappingOption() throws Exception {
...
 
 		OptionsContainer options = new ProgrammaticOptionValueSource( context ).getPropertyOptions( ExampleForMongoDBMapping.class, "content" );
 		assertThat( options.getUnique( AssociationStorageOption.class ) ).isEqualTo( AssociationStorageType.ASSOCIATION_DOCUMENT );
 	}
 
 	@SuppressWarnings("unused")
 	private static final class ExampleForMongoDBMapping {
+
 		String content;
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/options/associationstorage/AssociationStorageOptionTest.java]: why would you add this line?

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/options/writeconcern/WriteConcernPropagationTest.java [hunk 8]
```diff
@@ -114,7 +117,9 @@ public void shouldApplyConfiguredWriteConcernForRemoveTuple() {
 		session.close();
 
 		// then expect a call to remove with the configured write concern
-		verify( mockClient.getCollection( "GolfPlayer" ) ).remove( any( DBObject.class ), eq( WriteConcern.MAJORITY ) );
+		// verify( mockClient.getCollection( "GolfPlayer" ) ).remove( any( Document.class ), eq( WriteConcern.MAJORITY )
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/options/writeconcern/WriteConcernPropagationTest.java]: We typically don't comment out code, unless there's a very specific reason which might require an explanation comment

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/performance/MongoDBPerformanceTest.java [hunk 9]
```diff
@@ -88,7 +67,7 @@ public void testNumberOfCallsToDatastore() throws Exception {
 		assertThat( loadInvocationCount ).isEqualTo( 0 );
 		assertThat( updateInvocationCount ).isEqualTo( 0 );
 
-		//remove one of the elements and add a new one
+		// remove one of the elements and add a new one
```
Comment (Sanne) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/performance/MongoDBPerformanceTest.java]: An example of changes which should be avoided.

**Comment by schernolyas** [comment]
@Sanne Thank you for examples. I will fix it. I just fix formatting in classes by Eclipse formatter .... and broke formatting. I will restore it.

Can You download my branch  and review changes in API?

**Comment by Sanne** [comment]
> Can You download my branch and review changes in API?

I'll let @DavideD review those. But if you could fix the formatting first it would be a great help, as otherwise how can he figure out which changes do you mean?

**Comment by schernolyas** [comment]
@Sanne  Thanks!
I will fix formatting and tests. But .... changing is widescale really. 129 classes is not joke.

**Comment by schernolyas** [comment]
@Sanne  Thanks!
I will fix formatting and tests. But .... changing is widescale really. 129 classes is not joke.

**Comment by DavideD** [comment]
> ...129 classes is not joke.

The problem is that the number of classes that this commit should actually change is much lower.
The only reason so many classes have changes is because you squashed together in a single commit many unrelated fixes. The one I've spotted so far are:

* Renamed existing method. This might be acceptable, but this commit is already very big, why don't you make this kind of changes in a different one?
* Added the generic type in some places: again this is not necessarly wrong, but if this commit contains already so many changes why diing a new one?
* Change the formattation of some classes without any actual change in the code
* You added a new Annotation related to GridFS (@BinaryStorage) when we agreed that the commit should only contains changes related to removing the deprecated method

I'm going to close this PR for now and wait for a new tidier one.

Thanks
